test/e2e/scheduling/predicates.go

```diff
@@ -192,6 +194,138 @@ var _ = SIGDescribe("SchedulerPredicates [Serial]", func() {
+			if found && pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed {
+				framework.Logf("pod %q requesting resource cpu=%vm on node %q", pod.Name, getRequestedCPU(pod), pod.Spec.NodeName)
+				nodeToAllocatableMap[pod.Spec.NodeName] -= getRequestedCPU(pod)
+			}
+		}
+
+		ginkgo.By("Starting pods to consume most of the cluster CPU.")
+		// Create one pod per node that requires 70% of the node remaining CPU.
+		fillerPods := []*v1.Pod{}
+		for nodeName, cpu := range nodeToAllocatableMap {
+			requestedCPU := cpu * 7 / 10
+			framework.Logf("Creating a pod which consumes cpu=%vm on node %q", requestedCPU, nodeName)
+			fillerPods = append(fillerPods, createPausePod(f, pausePodConfig{
+				Name: "filler-pod-" + string(uuid.NewUUID()),
+				Resources: &v1.ResourceRequirements{
+					Limits: v1.ResourceList{
+						v1.ResourceCPU: *resource.NewMilliQuantity(requestedCPU, "DecimalSI"),
+					},
+					Requests: v1.ResourceList{
+						v1.ResourceCPU: *resource.NewMilliQuantity(requestedCPU, "DecimalSI"),
+					},
+				},
+				Affinity: &v1.Affinity{
+					NodeAffinity: &v1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      "node",
+											Operator: v1.NodeSelectorOpIn,
+											Values:   []string{nodeName},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}))
+		}
+		// Wait for filler pods to schedule.
+		for _, pod := range fillerPods {
+			framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(cs, pod), "failed to schedule pod %q", pod.Name)
+		}
+
+		// Register a runtimeClass with overhead set as 30% of the nodeMaxAllocatable
+		handler := e2enode.PreconfiguredRuntimeClassHandler(framework.TestContext.ContainerRuntime)
+
+		rc := &v1beta1.RuntimeClass{
+			ObjectMeta: metav1.ObjectMeta{Name: handler},
+			Handler:    handler,
+			Overhead: &v1beta1.Overhead{
+				PodFixed: v1.ResourceList{
+					v1.ResourceCPU: *resource.NewMilliQuantity(nodeMaxAllocatable*3/10, "DecimalSI"),
+				},
+			},
+		}
+
+		_, err = f.ClientSet.NodeV1beta1().RuntimeClasses().Create(rc)
+		framework.ExpectNoError(err, "failed to create RuntimeClass resource")
+
+		ginkgo.By("Creating another pod that requires unavailable amount of CPU.")
+		// Create another pod that requires 20% of the largest node CPU resources.
+		// This pod should remain pending as at least 70% of CPU of other nodes in
+		// the cluster are already consumed.
+		podName := "additional-pod"
+		conf := pausePodConfig{
+			RuntimeClassHandler: &handler,
+			Name:                podName,
+			Labels:              map[string]string{"name": "additional"},
+			Resources: &v1.ResourceRequirements{
+				Limits: v1.ResourceList{
+					v1.ResourceCPU: *resource.NewMilliQuantity(nodeMaxAllocatable*2/10, "DecimalSI"),
```

Comment:
To make this test non-disruptive to other running workloads or tests, we can test on a particular "fake" resource, so that this "fake" resource is used by this test only.

An example here: https://github.com/kubernetes/kubernetes/blob/652f52b51c05befece676f87d0ca8bafeded269d/test/e2e/scheduling/preemption.go#L276-L278

Reply:
Great pattern, will do this on the first Ready node available.

Reply:
Done